netty/src/main/java/io/grpc/transport/netty/Utils.java

```diff
@@ -173,8 +177,22 @@ private static Http2Headers convertMetadata(Metadata headers) {
 
     @Override
     public EventLoopGroup create() {
-      return new NioEventLoopGroup(nEventLoops, new ThreadFactoryBuilder().setNameFormat(name + "-%d")
-          .build());
+      ThreadFactory threadFactory = new ThreadFactoryBuilder().setNameFormat(name + "-%d").build();
+      try {
+        Constructor<NioEventLoopGroup> constructor = NioEventLoopGroup.class.getConstructor(
+            Integer.TYPE, ThreadFactory.class);
+        if (constructor == null) {
+          // is netty 5
+          constructor = NioEventLoopGroup.class.getConstructor(Integer.TYPE, Executor.class);
+          return constructor.newInstance(nEventLoops,
+              Executors.newFixedThreadPool(nEventLoops, threadFactory));
```

Comment:
@louiscryan if we pass in the `Executor` to the `EventLoopGroup`, we should also shut it down in close().